File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost.sort()
    reserve.sort()
    temp = reserve.copy()
    
    for i in temp:
        if i in lost:
            reserve.remove(i)
            lost.remove(i)
    
    for i in reserve:
        if i - 1 in lost:
            lost.remove(i-1)
        elif i + 1 in lost:
            lost.remove(i+1)
        
        
    return n - len(lost)

File: 프로그래머스/1/42862. 체육복/체육복.go
import "sort"

func solution(n int, lost []int, reserve []int) int {
    sort.Ints(lost)
    sort.Ints(reserve)

    lostMap := make(map[int]bool)
    reserveMap := make(map[int]bool)

    for _, l := range lost {
        lostMap[l] = true
    }
    for _, r := range reserve {
        reserveMap[r] = true
    }

    for student := range lostMap {
        if reserveMap[student] {
            delete(lostMap, student)
            delete(reserveMap, student)
        }
    }

    lostList := []int{}
    for student := range lostMap {
        lostList = append(lostList, student)
    }
    sort.Ints(lostList)

    for _, student := range lostList {
        if reserveMap[student-1] {
            delete(reserveMap, student-1)
            delete(lostMap, student)
        } else if reserveMap[student+1] {
            delete(reserveMap, student+1)
            delete(lostMap, student)
        }
    }

    return n - len(lostMap)
}
